src/cargo/ops/cargo_new.rs

```diff
@@ -811,6 +811,8 @@ fn main() {
             b"\
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
```

Comment:
I'm a bit mixed about how we solve the original concern.  I've posted my thoughts to https://github.com/rust-lang/cargo/issues/10559